src/core/common/settings.hpp

```diff
@@ -100,6 +100,7 @@ class SettingsBase : public InstanceLocator
         uint32_t        mPreviousPartitionId;    ///< PartitionId
         Mac::ExtAddress mExtAddress;             ///< Extended Address
         uint8_t         mMlIid[OT_IP6_IID_SIZE]; ///< IID from ML-EID
+        uint16_t        mVersion;                ///< Version
```

Comment:
Where is this used? I think this would be useful to detect if the device itself is changing versions. In this case, the device should remove any existing child/parent info and perform a full reattach.